components/storage/blockdevice/COMPONENT_FLASHIAP/FlashIAPBlockDevice.h

```diff
@@ -31,7 +31,11 @@ class FlashIAPBlockDevice : public BlockDevice {
     /** Creates a FlashIAPBlockDevice **/
     FlashIAPBlockDevice();
 
-    MBED_DEPRECATED("Please use default constructor instead")
+    /** Creates a FlashIAPBlockDevice
+     *
+     *  @address        Physical address where the block device start
+     *  @size           The block device size
+     */
     FlashIAPBlockDevice(uint32_t address, uint32_t size = 0);
```

Comment:
Why ? can you add details to the commit message why this is not deprecated anymore? And also changes in FlashIAPBlockDevice ctor  - part of the revert or?

Reply:
We originally deprecated this constructor in the external repo, as this would have required the user to either keep a list of absolute addresses or to use `FlashIAP` driver to query the flash parameters (which would have been "ugly" if we already use `FlashIAPBlockDevice`). However, it turned out that this had actually broke those who used this block device and finally reverted that change in the external repo. In addition, there was no escape from using `FlashIAP` to query flash parameters before instantiating `FlashIAPBlockDevice`, as we needed to query sector sizes in order to calculate the partial file system location in flash.

Reply:
Can you add some of these details to the commit msg?